tomatic/callinfo.py

```diff
@@ -105,24 +105,17 @@ def partnerInfo(self, partner_data):
         return result
 
     def getPartnerRelatedContracts(self, partner_id):
-        contract_tp_ids = self.O.GiscedataPolissa.search([
-            ('titular.id', '=', partner_id),
-            ('pagador.id', '=', partner_id),
-            ])
         contract_titular_ids = self.O.GiscedataPolissa.search([
             ('titular.id', '=', partner_id),
-            ])
+        ])
```

Comment:
I didn't check the pr thoroughly as i am in holidays. I gues you are simplifying the query again because you put the sorting anywhere else. Cool. You can further speed this by joining the three queries into one joining all three conditions with an or. Three is better than four but one is better than three and you can avoid all the later post processing to join the lists.